src/gps/RTC.cpp

```diff
@@ -39,22 +39,23 @@ bool perhapsSetRTC(RTCQuality q, const struct timeval *tv)
         currentQuality = q;
         shouldSet = true;
         DEBUG_MSG("Upgrading time to RTC %ld secs (quality %d)\n", tv->tv_sec, q);
-    } else if(q == RTCQualityGPS && (now - lastSetMsec) > (12 * 60 * 60 * 1000L)) {
-        // Every 12 hrs we will slam in a new GPS time, to correct for local RTC clock drift
+    } else if(q > RTCQualityNone && q == currentQuality && (now - lastSetMsec) > (12 * 60 * 60 * 1000L)) {
+        // Every 12 hrs we will slam in a new time, to correct for local RTC clock drift
```

Comment:
hmm - what was the reason for changing this test?  It would seem to slam in clock values that we've recevied from the rest of the net (RTCQualityFromNet).  Don't we only want to do that if we have a known 'gold standard' local GPS time?

Reply:
The changed code updates clock of the same quality (the "q == currentQuality") only, so it will not replace GPS clock with Net clock. The idea is to periodically update GPS-less devices from the net clocks. Coming to think about it now, it seems that it is more trouble than it is worth - I would probably want a different update period in this case, etc. Reverted this change.